logic.py

```diff
@@ -109,3 +112,69 @@ def get_task_list_with_milestones(dataset_id, tasks, milestones):
             })
             last_milestone_id = milestone_id
     return task_list_with_milestones
+
+
+def complete_task(workflow_state, task_id):
+    new_state = copy.deepcopy(workflow_state)
+    completed_tasks = set(task['id'] for task in new_state["completedTasks"])
+    if task_id not in completed_tasks:
+        new_state["completedTasks"].append({
+            "id": task_id,
+            "completedAt": datetime.now(timezone.utc)
+        })
+    return new_state
+
+
+def uncomplete_task(workflow_state, task_id):
+    new_state = copy.deepcopy(workflow_state)
+    completed_tasks = {task['id']: task for task in new_state["completedTasks"]}
+    if task_id not in completed_tasks:
+        raise LogicError(f"Task {task_id} not found in completed tasks")
+    new_state["completedTasks"].remove(completed_tasks[task_id])
+    return new_state
+
+
+def check_if_task_is_complete(task_id, wf_state):
+    completed_tasks = {task['id']: task for task in wf_state["completedTasks"]}
```

Comment:
Why not just a plain set?

```suggestion
    completed_tasks = {task['id'] for task in wf_state["completedTasks"]}
```